framework/tlvf/AutoGenerated/include/tlvf/ieee_1905_1/sVendorOUI.h

```diff
@@ -0,0 +1,45 @@
+///////////////////////////////////////
+// AUTO GENERATED FILE - DO NOT EDIT //
+///////////////////////////////////////
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _SUINT_24_t_H_
+#define _SUINT_24_t_H_
+
+#include <stdint.h>
+
+typedef struct sVendorOUI {
```

Comment:
Strict name should be uint_24, shouldn't it? We might need it in other places.

Reply:
if it's called `uint24_t` then tlvf will recognize it as an std type, which will cause it not to call `struct_swap()`

Reply:
if it's called `uint24_t`, tlvf will recognize it as an std type, and won't call `struct_swap` as needed

Reply:
Can't we change this so that tlvf will have a map for the simple types? and then not put uint24_t in the simple types map?

Reply:
In its current form, it is not a uint24_t (a 24-bit integer) but it is indeed an OUI structure (a 3-byte value with an upper, middle and lower byte). If we consider it a uint24_t, then it should just have conversions from/to uint32_t and a swap() function. But actually I think the sVendorOUI is good. Of course, then the file name is not correct.